ibis/tests/test_impala_e2e.py

```diff
@@ -903,6 +903,179 @@ def test_sql_query_limits(self):
+        assert result == val
+
+    def test_float_wrapping(self):
+        symbol = 'Identity'
+        name = 'identity'
+        inputs = ['float']
+        output = 'float'
+        udf_info = udf.UDFCreator(ENV.udf_so, inputs, output, symbol, name)
+        self.con.create_udf(udf_info, db=self.test_data_db)
+        op = udf_info.to_operation()
+        udf.add_impala_operation(op, name, self.test_data_db)
+        assert self.con.exists_udf(name, self.test_data_db)
+
+        def _identity_test(value):
+            return op(value).to_expr()
+        val = ibis.literal(3.14)
+        expr = _identity_test(val)
+        result = self.con.execute(expr)
+        assert result == val
+
+    def test_double_wrapping(self):
+        symbol = 'Identity'
+        name = 'identity'
+        inputs = ['double']
+        output = 'double'
+        udf_info = udf.UDFCreator(ENV.udf_so, inputs, output, symbol, name)
+        self.con.create_udf(udf_info, db=self.test_data_db)
+        op = udf_info.to_operation()
+        udf.add_impala_operation(op, name, self.test_data_db)
+        assert self.con.exists_udf(name, self.test_data_db)
+
+        def _identity_test(value):
+            return op(value).to_expr()
+        val = ibis.literal(3.14)
+        expr = _identity_test(val)
+        result = self.con.execute(expr)
+        assert result == val
+
+    def test_string_wrapping(self):
+        symbol = 'Identity'
+        name = 'identity'
+        inputs = ['string']
+        output = 'string'
+        udf_info = udf.UDFCreator(ENV.udf_so, inputs, output, symbol, name)
+        self.con.create_udf(udf_info, db=self.test_data_db)
+        op = udf_info.to_operation()
+        udf.add_impala_operation(op, name, self.test_data_db)
+        assert self.con.exists_udf(name, self.test_data_db)
+
+        def _identity_test(value):
+            return op(value).to_expr()
+        val = ibis.literal('ibis')
+        expr = _identity_test(val)
+        result = self.con.execute(expr)
+        assert result == val
+
+    @pytest.mark.xfail
+    def test_timestamp_wrapping(self):
+        symbol = 'Identity'
+        name = 'identity'
+        inputs = ['timestamp']
+        output = 'timestamp'
+        udf_info = udf.UDFCreator(ENV.udf_so, inputs, output, symbol, name)
+        self.con.create_udf(udf_info, db=self.test_data_db)
+        op = udf_info.to_operation()
+        udf.add_impala_operation(op, name, self.test_data_db)
+        assert self.con.exists_udf(name, self.test_data_db)
+
+        def _identity_test(value):
+            return op(value).to_expr()
+        val = ibis.literal(1000)
+        expr = _identity_test(val)
+        result = self.con.execute(expr)
+        assert result == val
+
+    def test_mult_type_args_wrapping(self):
+        symbol = 'AlmostAllTypes'
+        name = 'all_types'
+        inputs = ['string', 'boolean', 'tinyint', 'smallint', 'int',
+                  'bigint', 'float', 'double']
```

Comment:
Let's standardize on Ibis type names from the user's side. So `int16` instead of `smallint`, etc. We can put a table in the documentation for Impala developers